frontend/src/components/chatInput/ChatInput.tsx

```diff
@@ -1,30 +1,42 @@
 import {
-	Form,
-	InputWrapper,
-	MemeButton,
-	MemeGenerateIcon,
-	MessageInput
+    Form,
+    InputWrapper,
+    MemeButton,
+    MemeGenerateIcon,
+    MessageInput
 } from "./ChatInput.styled";
 import useChatInput from "./useChatInput";
+import GifWidget from "components/GifWidget/GifWidget";
+import { useState } from "react";
 
 const ChatInput = () => {
-	const { currentInputValue, handleSetCurrentInputValue, handleSubmit } =
-		useChatInput();
+    const { currentInputValue, handleSetCurrentInputValue, handleSubmit } =
+        useChatInput();
+    const [gifWidgetVisible, setGifWidgetVisible] = useState(false);
 
-	return (
-		<InputWrapper>
-			<Form onSubmit={handleSubmit}>
-				<MessageInput
-					onChange={handleSetCurrentInputValue}
-					value={currentInputValue}
-					autoFocus
-				/>
-			</Form>
-			<MemeButton>
-				<MemeGenerateIcon />
-			</MemeButton>
-		</InputWrapper>
-	);
+    const toggleGifWidgetVisibility = () => {
+        setGifWidgetVisible(prevState => !prevState);
+    };
+
+    const GifWidgetContainer = () => {
+        return gifWidgetVisible ? <GifWidget /> : null;
```

Comment:
This could be simplify, use it inline on "return" and create it like that:

```
gifWidgetVisible && <GifWidget/>
```

Reply:
done